x/supplier/client/cli/stake_options_reader.go

```diff
@@ -0,0 +1,157 @@
+package cli
+
+import (
+	"fmt"
+	"net/url"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"gopkg.in/yaml.v2"
+
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+)
+
+// ParsedStakeConfig is a parsed version of the stake config file
+type ParsedStakeConfig struct {
+	stake    sdk.Coin
+	services []*sharedtypes.SupplierServiceConfig
+}
+
+// StakeConfig is the structure of the stake config file
+type StakeConfig struct {
```

Comment:
Confirming that these structs, and their underlying types, need to be exposed publically?

Reply:
1. (@red-0ne) I think this should be moved to a different package; while it's on the boundary between CLI and library code, I am going to argue that it's more useful (better DevEx) to categorize it as library code. Perhaps `x/supplier/client` or `x/supplier/client/config`.
2. I think it's reasonable to expose config structs as part of the API to provide programatic usage. 
3. Following-on from 1, I think it would then also be necessary to export these structs.
4. Regardless of 1-3, it is necessary to export the fields on these types in order for them to be (de)serializable.